### PR TITLE
Add sprite fades for occluding sprites

### PR DESCRIPTION
**Pull request by metalgearsloth**
Prevents player entity from being completely occluded by designated entities. This also makes it less annoying to pickup stuff under trees as you don't need to right-click and read through everything.

https://user-images.githubusercontent.com/31366439/210167980-c88fc4d8-b05f-434e-b4f2-63fd610a9dbb.mp4

:cl:
- add: Trees will now fade out slightly instead of completely occluding you if you are under them.

### CODE REVIEW

**Comment by Pangogie** [comment]
Neat!

**Comment by mirrorcult** [comment]
this shit way too fancy for our dinky game

File: Content.Client/Sprite/SpriteFadeSystem.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        if (_stateManager.CurrentState is not GameplayState state)
+            return;
+
+        var spriteQuery = GetEntityQuery<SpriteComponent>();
+        var fadeQuery = GetEntityQuery<SpriteFadeComponent>();
+        var change = ChangeRate * frameTime;
+        var mapPos = playerXform.MapPosition;
+
+        // Also want to handle large entities even if they may not be clickable.
+        foreach (var ent in state.GetEntitiesUnderPosition(mapPos, 6f))
```
Comment (ElectroJr) [inline Content.Client/Sprite/SpriteFadeSystem.cs:56]: Contrary to the comment, this only returns clickable entities (though maybe without an interaction outline).  If you want to include non-clickable entities, then clickable & click-maps should be separated out.

**Comment by Emisse** [comment]
does this work for entities that go under tables and doors and the like

**Comment by metalgearsloth** [comment]
> does this work for entities that go under tables and doors and the like

The component has to be added them but it can.

**Pull request merged**